### PR TITLE
fix: avoid gRPC FTS file modification

### PR DESCRIPTION
**Pull request by hpohekar**
closes https://github.com/ansys/pyfluent/issues/2735

Do not upload file if it's already exists on remote.

### CODE REVIEW

File: src/ansys/fluent/core/utils/file_transfer_service.py [hunk 1]
```diff
@@ -229,7 +234,12 @@ def upload(
         files = _get_files(file_name, self.host_mount_path)
         if self.client:
             for file in files:
-                if os.path.isfile(file):
+                is_file_exists_on_remote = self.file_exists_on_remote(
+                    os.path.basename(file)
+                )
+                if is_file_exists_on_remote:
+                    print(f"\n{file} already uploaded.\n")
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/utils/file_transfer_service.py:241]: Shall we use the `warnings` module? 

A more accurate message would say that a file with the *same name* *exists at the remote location*.   We don't know if it's the same file or if it were uploaded.

File: src/ansys/fluent/core/utils/file_transfer_service.py [hunk 2]
```diff
@@ -229,7 +234,12 @@ def upload(
         files = _get_files(file_name, self.host_mount_path)
         if self.client:
             for file in files:
-                if os.path.isfile(file):
+                is_file_exists_on_remote = self.file_exists_on_remote(
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/utils/file_transfer_service.py:237]: Either leave off the `is_` part or rename it `does_file_exist_on_remote` (which is better grammatically but too long and clumsy) or just `is_file_on_remote`.

Comment (hpohekar) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: Right, `is_file_on_remote` is correct and looks good. Done. Thank you.

File: src/ansys/fluent/core/utils/file_transfer_service.py [hunk 3]
```diff
@@ -229,7 +234,12 @@ def upload(
...
+                    os.path.basename(file)
+                )
+                if is_file_exists_on_remote:
+                    print(f"\n{file} already uploaded.\n")
+                elif os.path.isfile(file) and not is_file_exists_on_remote:
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/utils/file_transfer_service.py]: According to my understanding, this aligns the gRPC file transfer service behaviour with the PIM based behaviour that's (currently) implemented in a separate class. From that point of view, we can merge this now, and then consider what the overall behaviour should be and to what extent it should be configurable by the user.

File: src/ansys/fluent/core/utils/file_transfer_service.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: In whole source code we are not using warning module to check the existing files. `(examples.download(), PIMFileTransferService,)` Therefore we are not using it here. We have updated the message now. Thank you.

File: src/ansys/fluent/core/utils/file_transfer_service.py (hunk 3 above)
Comment (hpohekar) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: Yes, sure.

File: src/ansys/fluent/core/utils/file_transfer_service.py (hunk 1 above)
Comment (seanpearsonuk) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: I'm not sure about using print. One thing about using `warnings.warn()` is that allows specific message types to be disabled.

Comment (hpohekar) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: Right, I agree. We will update this now. Thank you.

Comment (hpohekar) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: Done. Thank you.

Comment (seanpearsonuk) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: Overall, as this is a library, we must be careful about writing to stdout.

Comment (hpohekar) [reply src/ansys/fluent/core/utils/file_transfer_service.py]: Yes, I agree.

**Comment by mkundu1** [comment]
@hpohekar I think, in general, we still want to upload the file when it exists in the server as the client side file can be newer. A better check here may be not to upload the file if the local and remote absolute filepaths are identical. Nevertheless, feel free to merge the current PR to fix the immediate issues in the regressions, we can consider this in a later PR. CC: @seanpearsonuk

**Review by mkundu1 — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

**Pull request merged**